syspage.c

```diff
@@ -518,9 +516,7 @@ int syspage_addmap(const char *name, addr_t start, addr_t end, const char *attr)
 	if (syspage_strAttr2ui(attr, &map->attr) < 0)
 		return -EINVAL;
 
-	len = hal_strlen(name);
-	size = min(len, SIZE_MAP_NAME - 1);
-
+	size = min(hal_strlen(name), SIZE_MAP_NAME - 1);
```

Comment:
In your change `hal_strlen` is invoked twice in `min`. It was the main reason why there is added `len`.
Yeah, your change remove one line of code, but I don't think that it is a better approach.

Reply:
Thanks for the explanation. I didn't realize that min macro would force hal_strlen() evaluation twice. I restored the original version.